modules/renter/hostdb/hostdb.go

```diff
@@ -52,31 +51,27 @@ type HostDB struct {
 
 	// The hostTree is the root node of the tree that organizes hosts by
 	// weight. The tree is necessary for selecting weighted hosts at
-	// random. 'activeHosts' provides a lookup from hostname to the the
-	// corresponding node, as the hostTree is unsorted. A host is active if
-	// it is currently responding to queries about price and other
-	// settings.
-	hostTree    *hosttree.HostTree
-	activeHosts map[modules.NetAddress]*hostEntry
-
-	// allHosts is a simple list of all known hosts by their network address,
-	// including hosts that are currently offline.
-	allHosts map[modules.NetAddress]*hostEntry
+	// random.
+	hostTree *hosttree.HostTree
 
 	// the scanPool is a set of hosts that need to be scanned. There are a
 	// handful of goroutines constantly waiting on the channel for hosts to
 	// scan.
-	scanList []*hostEntry
-	scanPool chan *hostEntry
+	scanList []modules.HostDBEntry
+	scanPool chan modules.HostDBEntry
 	scanWait bool
+	online   bool
 
 	blockHeight types.BlockHeight
 	lastChange  modules.ConsensusChangeID
 }
 
 // New returns a new HostDB.
-func New(cs consensusSet, persistDir string) (*HostDB, error) {
+func New(g modules.Gateway, cs consensusSet, persistDir string) (*HostDB, error) {
```

Comment:
I really don't like that style of dependency management, because it adds a lot of annoyance when you decide that you want some other gateway function as well. We don't want an interface that can tell us what peers we have, we want the whole entire gateway that does all the gateway things. Some alternate interface which has a Peers() method but isn't a gateway (maybe it's a kadmelia implementation or something) wouldn't be the same, because it doesn't have all the same standards for aggressively hunting non-local outbound peers, and might not use the same keepalive algorithms.